### PR TITLE
feat: Remove unused code related to nginx

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
We don't use nginx for redirects. Redirects are handled through: https://docs.sentry.io/contributing/pages/redirects/

So in this PR, unused code related to nginx and its' redirects should be removed.

cc: @BYK since we are not generating a nginx.out.conf file, I removed it from the Dockerfile

Consensus is that the Dockerfile is not necessary, so it will be also removed in this PR.

## Test Plan
Check the Vercel preview build to make sure redirects did not break.

### CODE REVIEW

**Comment by dcramer** [comment]
tbqh we should remove the dockerfile and everything else that I forgot to clean up from pre-vercel era

**Review by dcramer — APPROVED** [review]

**Review by dcramer — CHANGES_REQUESTED** [review]
wait you cant remove the nginx stuff from gatsby, thats used by vercel

**Comment by NisanthanNanthakumar** [comment]
> wait you cant remove the nginx stuff from gatsby, thats used by vercel

The preview build's redirects seem to be working

**Comment by dcramer** [comment]
its because you're seeing the client side redirects - the nginx.conf thing is a vercel compatible redirect solution as far as I remember. it's possible I'm making that up but i would verify that its actually literally unused (curl the page you should see a 302 not a 200).

**Comment by NisanthanNanthakumar** [comment]
> its because you're seeing the client side redirects - the nginx.conf thing is a vercel compatible redirect solution as far as I remember. it's possible I'm making that up but i would verify that its actually literally unused (curl the page you should see a 302 not a 200).

Screen Shot 2020-09-28 at 2 33 23 PM

@dcramer I checked against the preview build and current live docs and both have the same results.

**Comment by dcramer** [comment]
deferring here as its possible i dont remember correctly

File: Dockerfile [hunk 1]
```diff
@@ -29,7 +29,4 @@ RUN gatsby build
 
 FROM gatsbyjs/gatsby
```
Comment (BYK) [inline Dockerfile:30]: We should just remove this file, it is not used by anything.

File: src/gatsby/config.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import axios from "axios";
 
 import queries from "./utils/algolia";
-import PackageRegistry from "./utils/packageRegistry";
+// import PackageRegistry from "./utils/packageRegistry";
```
Comment (BYK) [inline src/gatsby/config.ts:4]: No commented-out code please. Either delete or uncomment.

File: src/gatsby/config.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ if (process.env.DISABLE_THUMBNAILS === "1") {
 
 const getPlugins = () => {
   const remarkPlugins = [
-    {
+   /* {
```
Comment (BYK) [inline src/gatsby/config.ts:20]: Same as above.

**Review by BYK — CHANGES_REQUESTED** [review]
Thanks for the clean up @NisanthanNanthakumar!

@dcramer don't we use `vercel.json` now for all redirects: https://github.com/getsentry/sentry-docs/blob/master/vercel.json

Rejecting for commented-out code and for the removal of `Dockerfile`, LGTM otherwise.

**Comment by BYK** [comment]
There are some link failures here: https://github.com/getsentry/sentry-docs/runs/1181514785?check_suite_focus=true

Should we be adding more redirects to the `vercel.json` file?

File: src/gatsby/config.ts (hunk 2 above)
Comment (dcramer) [reply src/gatsby/config.ts:4]: guessing this was an accident - this code cannot be removed

Comment (NisanthanNanthakumar) [reply src/gatsby/config.ts:4]: whoops! this is my workaround to make gatsby build faster locally. I'm going to undo.

File: Dockerfile (hunk 1 above)
Comment (BYK) [reply Dockerfile:30]: Let's remove this file as a whole as w/o Nginx config this is useless.

**Comment by BYK** [comment]
Linkchecker does not fail on master so I think those issues are somehow related to this change. Can you look into it as I don't feel confident with merging in this state?

**Review by BYK — CHANGES_REQUESTED** [review]
See my comment above.

**Review by BYK — APPROVED** [review]
Great job, thanks a lot @NisanthanNanthakumar!

**Pull request merged**